internal/cli/cmd/backup/list.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package backup
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	dataprotectionv1alpha1 "github.com/apecloud/kubeblocks/apis/dataprotection/v1alpha1"
+	"github.com/apecloud/kubeblocks/internal/cli/list"
+	"github.com/apecloud/kubeblocks/internal/cli/printer"
+	"github.com/apecloud/kubeblocks/internal/cli/types"
+	"github.com/apecloud/kubeblocks/internal/cli/util"
+	"github.com/apecloud/kubeblocks/internal/constant"
+)
+
+var (
+	listBackupExample = templates.Examples(`
+		# list all backups
+		kbcli backup list
+
+		# list all backups of specified cluster
+		kbcli backup list --cluster mycluster
+	`)
+)
+
+type backupListOptions struct {
+	*list.ListOptions
+	cluster string
+}
+
+func newListCommand(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &backupListOptions{ListOptions: list.NewListOptions(f, streams, types.BackupGVR())}
+	cmd := &cobra.Command{
+		Use:               "list",
+		Short:             "List backups.",
+		Aliases:           []string{"ls"},
+		Example:           listBackupExample,
+		ValidArgsFunction: util.ResourceNameCompletionFunc(f, o.GVR),
+		Run: func(cmd *cobra.Command, args []string) {
+			o.Names = args
+			util.CheckErr(backupListRun(o))
+		},
+	}
+	o.AddFlags(cmd, true)
+	cmd.Flags().StringVar(&o.cluster, "cluster", "", "List backups in the specified cluster")
+	util.RegisterClusterCompletionFunc(cmd, f)
+
+	return cmd
+}
+
+func backupListRun(o *backupListOptions) error {
```

Comment:
Better to reuse the code of `kbcli cluster list-backups` in https://github.com/apecloud/kubeblocks/blob/4a4f269452ff72aecb347c986c7ddbcb15c2df47/internal/cli/cmd/cluster/dataprotection.go#L268-L268

Reply:
If reuse the code of `kbcli cluster list-backups`, I can't achieve this, list multiple backups with `kbcli backup list a b`.
![image](https://github.com/apecloud/kubeblocks/assets/49025948/02337299-b130-44a4-9e22-c4498abff776)